src/generic_core/remote-instance.ts

```diff
@@ -148,24 +163,90 @@ module Core {
      * TODO: return a boolean on success/failure
      */
     public handleSignal = (type:uProxy.MessageType,
-                           signalFromRemote:Object) => {
+                           signalFromRemote:Object) :Promise<void> => {
       if (uProxy.MessageType.SIGNAL_FROM_CLIENT_PEER === type) {
         // If the remote peer sent signal as the client, we act as server.
         if (!this.user.consent.localGrantsAccessToRemote) {
           log.warn('Remote side attempted access without permission');
-          return;
+          return Promise.resolve<void>();
         }
 
-        // Create a new rtcToNet object everytime there is an OFFER signal
-        if(signalFromRemote['type'] == WebRtc.SignalType.OFFER) {
-          this.connection_.startShare();
+        // Create a new rtcToNet object everytime there is an OFFER signal.
+        if (signalFromRemote['type'] == WebRtc.SignalType.OFFER) {
+          this.connection_.resetRtcToNetCreated();
```

Comment:
Rather than call resetRtcToNetCreated here in remote-instance, can it be called in remote-connection's startShare method instead?  It seems like an implementation detail of RemoteConnection that is better kept hidden from RemoteInstance